src/formatting/formatter.rs

```diff
@@ -392,20 +432,76 @@ impl Formatter for FixFormatter {
             Value::Icon(_) => Err(Error::new_format(
                 "An icon cannot be formatted with 'fix' formatter",
             )),
+            Value::Timestamp(..) => Err(Error::new_format(
+                "A timestamp cannot be formatted with 'fix' formatter",
+            )),
             Value::Flag => Err(Error::new_format(
                 "A flag cannot be formatted with 'fix' formatter",
             )),
         }
     }
 }
 
+#[derive(Debug)]
+pub struct TimestampFormatter {
+    items: Option<Vec<Item<'static>>>,
+}
+
+fn make_static_item(item: Item<'_>) -> Item<'static> {
+    match item {
+        Item::Literal(str) => Item::OwnedLiteral(str.into()),
+        Item::OwnedLiteral(boxed) => Item::OwnedLiteral(boxed),
+        Item::Space(str) => Item::OwnedSpace(str.into()),
+        Item::OwnedSpace(boxed) => Item::OwnedSpace(boxed),
+        Item::Numeric(numeric, pad) => Item::Numeric(numeric, pad),
+        Item::Fixed(fixed) => Item::Fixed(fixed),
+        Item::Error => Item::Error,
+    }
+}
+
+impl TimestampFormatter {
+    fn new(format: Option<String>, locale: Option<String>) -> Result<Self> {
+        let format = format.as_deref().unwrap_or(DEFAULT_DATETIME_FORMAT);
+
+        let items = match locale.as_deref() {
+            Some(locale) => {
+                let locale = locale.try_into().ok().error("invalid locale")?;
+                StrftimeItems::new_with_locale(format, locale)
+            }
+            None => StrftimeItems::new(format),
+        }
+        .map(make_static_item)
+        .collect();
+
+        Ok(Self { items: Some(items) })
+    }
+}
+
+impl Formatter for TimestampFormatter {
+    fn format(&self, val: &Value) -> Result<String> {
+        let items = self.items.clone().unwrap();
+        match val {
+            Value::Number { .. } | Value::Text(_) | Value::Icon(_) | Value::Flag => unreachable!(),
```

Comment:
This code is not actually `unreachable`. The reason this is true for `FlagFormatter` is because there is no way for user to explicitly specify flag formatter.

Reply:
The good news is that this code can be simplified by implementing `Value::type_name(&self) -> &'static str`:

```rust
match val {
    Value::Timestamp(timezone) => todo!(),
    other => Err(Error::new_format(format!(
        "{} cannot be formatted with 'datetime' formatter",
        other.type_name(),
    ))),
}
```

Reply:
```rs
Err(Error::new_format(format!(
    "{:?} cannot be formatted with 'datetime' formatter",
    other,
)))
```

Also seems to work

Reply:
The problem with this is that it prints the internals of the `Value`, which is an implementation detail and kind off irrelevant, making the error message longer that it needs to be.